### PR TITLE
feat(create-rslib): add react template and tools(storybook/vitest)

### PR DESCRIPTION
**Pull request by fi3ework**
#### Programmatically generate templates

- The generated template path is in pattern of `[${template}]-[${(tools ? Object.keys(tools) : []).sort()}]-${lang}`.
- Add new tools step (support vitest and storybook for now), `vitest` is applicable for all templates and `storybook` is applicable for UI related template (react).

#### Add React template

- React (and other UI frameworks) is applicable with tool Storybook.

#### Templates tweak

- Adding `typescript` dependency and `dts: true` for JS projects: Emitting dts for JS templates. It's acceptable to write non-TS project sometimes, but dts is recommended to emit and enable partial type check when needed.

#### TODO

- As the templates getting more complex, I wonder does the `tools` step merge into `create-rstack`?
- Is adding TS stuff to JS project also applicable for Rsbuild JS projects?

## Checklist

- [x] Tests updated (or not required).
- [ ] Documentation updated (or not required).

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -1 +1 @@
-v18.20.2
```
Comment (Timeless0911) [inline .nvmrc:1]: Will this affect the CI environment? I don't think it's necessary to use node version too new here.

File: packages/create-rslib/fragments/base/react-js/rslib.config.mjs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+export default defineConfig({
+  source: {
+    entry: {
+      index: ['./src/**'],
+    },
+  },
+  lib: [
+    {
+      ...shared,
```
Comment (Timeless0911) [inline packages/create-rslib/fragments/base/react-js/rslib.config.mjs:19]: no need to use shared here?

File: packages/create-rslib/fragments/base/react-js/rslib.config.mjs [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import { pluginReact } from '@rsbuild/plugin-react';
+import { defineConfig } from '@rslib/core';
+
+const shared = {
+  bundle: false,
+  dts: {
```
Comment (Timeless0911) [inline packages/create-rslib/fragments/base/react-js/rslib.config.mjs:6]: we can use `dts: true`

File: packages/create-rslib/fragments/base/react-ts/rslib.config.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
+import { pluginReact } from '@rsbuild/plugin-react';
+import { pluginSass } from '@rsbuild/plugin-sass';
+import { type LibConfig, defineConfig } from '@rslib/core';
+
+const shared: LibConfig = {
+  bundle: false,
+  dts: {
```
Comment (Timeless0911) [inline packages/create-rslib/fragments/base/react-ts/rslib.config.ts:7]: `dts: true`

File: packages/create-rslib/fragments/base/react-ts/rslib.config.ts [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    entry: {
+      index: ['./src/**', '!./src/env.d.ts'],
+    },
+  },
+  lib: [
+    {
+      ...shared,
+      format: 'esm',
+      output: {
+        distPath: {
```
Comment (Timeless0911) [inline packages/create-rslib/fragments/base/react-ts/rslib.config.ts:23]: The templated package of `node-dual` not set different output folder for esm/cjs, and it should be same with specification in package.json

File: packages/create-rslib/fragments/tools/vitest-node-ts/package.json [hunk 6]
```diff
@@ -0,0 +1,8 @@
+{
+    "scripts": {
```
Comment (Timeless0911) [inline packages/create-rslib/fragments/tools/vitest-node-ts/package.json:2]: the indent size seem larger

File: .nvmrc (hunk 1 above)
Comment (fi3ework) [reply .nvmrc:1]: > Will this affect the CI environment?

No, we specify the node version there.

> I don't think it's necessary to use node version too new here.

For copying files recursively introduced in node 22. It only requires developer's local node version.

File: packages/create-rslib/fragments/base/react-js/rslib.config.mjs (hunk 2 above)
Comment (fi3ework) [reply packages/create-rslib/fragments/base/react-js/rslib.config.mjs:19]: The PR is still in the WIP stage.

File: packages/create-rslib/fragments/tools/vitest-node-ts/package.json (hunk 6 above)
Comment (fi3ework) [reply packages/create-rslib/fragments/tools/vitest-node-ts/package.json]: we lost format on stage before

File: packages/create-rslib/fragments/base/react-ts/rslib.config.ts [hunk 7]
```diff
@@ -0,0 +1,24 @@
+import { pluginReact } from '@rsbuild/plugin-react';
+import { type LibConfig, defineConfig } from '@rslib/core';
```
Comment (chenjiahan) [inline packages/create-rslib/fragments/base/react-ts/rslib.config.ts:2]: Unused `LibConfig`

**Comment by chenjiahan** [comment]
> Is adding TS stuff to JS project also applicable for Rsbuild JS projects?

I think developers of JS projects may not want to use TypeScript

**Comment by Timeless0911** [comment]
The PR title can be more precise🥹

**Comment by fi3ework** [comment]
> I think developers of JS projects may not want to use TypeScript

I was wrong, I forget TS can't only check part of the file. Will remove TS check.

**Review by chenjiahan — APPROVED** [review]

**Comment by fi3ework** [comment]
The generated template files are numerous and may continue to grow. We can generate these template files during the create-rslib `publish` or `postinstall` stages. However, we need more testing to ensure this approach works. For now, we will keep the template files in the repository to assist with code reviews.

**Pull request merged**